Framework/API/src/FileFinder.cpp

```diff
@@ -470,14 +474,26 @@ const API::Result<std::string> FileFinderImpl::findRun(const std::string &hintst
 
   auto path = getPath(archs, filenames, uniqueExts);
   if (path) {
-    g_log.information() << "found path = " << path << '\n';
+    g_log.information() << "Found path = " << path << '\n';
     return path;
-  } else {
-    g_log.information() << "Unable to find run with hint " << hint << "\n";
   }
 
-  g_log.information() << "Unable to find file path for " << hint << "\n";
+  // Path not found
+  if (!useExtsOnly && uniqueExts.size() == 1) {
+    uniqueExts.pop_back(); // No need to search for missing extension again
+    getUniqueExtensions(exts, uniqueExts);
+    getUniqueExtensions(extensions, uniqueExts);
```

Comment:
Just in general I think having three vectors, `exts`, `extensions`, `uniqueExts`, is very confusing. If you can think of a way to rename them / restructure things, I think that'd be very helpful.

As far as I can see `extensions` is just assigned the contents of `facility_extensions` so I'm wondering if it even needs to exist? Why not just use `facility_extensions`? (and maybe rename it to `facilityExtensions` while you're at it)

I also think `findRun` is in need of a doc string, `userExtsOnly` is pretty ambiguous. None of this is your fault, just a good time to improve it.

Reply:
No, I agree fully, it is incredibly confusing. I'll make it a bit better to understand. You're right, extensions is just the facility extensions, `exts` is just the argument used with `useExtsOnly` and `uniqueExts` are the extensions that are effectively considered. I'll clear it up.